nncf/experimental/onnx/graph/model_transformer.py

```diff
@@ -140,16 +132,67 @@ def select_model_inputs_outputs(model, outputs=None):
                     raise RuntimeError
                 extra_model_outputs.add(edge_name)
             extra_model_outputs.update(input_edge_names)
-
-        model_with_intermediate_outputs = select_model_inputs_outputs(self.transformed_model,
-                                                                      outputs=[*extra_model_outputs,
-                                                                               *model_outputs])
-        self.transformed_model = model_with_intermediate_outputs
-
-    def _apply_quantizer_insertion_transformations(self) -> None:
+        return extra_model_outputs
+
+    def _insert_outputs(self, model: onnx.ModelProto, outputs: List[str] = None) -> onnx.ModelProto:
+        """
+        Takes a model and changes its outputs.
+
+        :param model: *ONNX* model
+        :param outputs: new outputs
```

Comment:
Can you clarify than is the format of the output name? Is is a name of the layer, or tensor name, etc.?

Reply:
Output names will be the same as the tensor names that they were placed on to collect data.
cc @kshpv

Reply:
outputs here are Edge names, which we should treat as model outputs

Reply:
@KodiaqQ please add this in docstring

Reply:
Done

Reply:
I think this was a missclick, right? I don't see the docstring with the description of outputs

Reply:
Not missclick. Check changes again.